### DISCUSSION TITLE
MPS GPU not being used when running run_localGPT.py

### DISCUSSION THREAD

**Discussion by Satyam7166-tech** [PromtEngineer/localGPT#430] [Q&A]
@PromtEngineer 

System: M1 pro
Model: TheBloke/Llama-2-7B-Chat-GGML

Here is my GPU usaage when I run ingest.py (with mps enabled)
<img width="1511" alt="mps-ingest" src="https://github.com/PromtEngineer/localGPT/assets/62897696/379d77d7-5682-4a43-ac37-d04447c91985">

And now look at the GPU usage when I run run_localGPT.py (with mps enabled)
<img width="1504" alt="mps-run-1" src="https://github.com/PromtEngineer/localGPT/assets/62897696/22460272-5a54-40f3-bc51-e9b65ee6519a">
 
 The spike is very thick (ignore the previous thick spike. It is denoting ingest) and happens just about 2 seconds before the LLM generates the answer.
And here is the message specifying that mps is being used.

<img width="854" alt="image" src="https://github.com/PromtEngineer/localGPT/assets/62897696/b00ce691-9682-496c-a8c1-540bff258672">

I am a complete beginner and any input will be greatly appreciated.
Thanks

**Comment by PromtEngineer** [comment]
When you run this, is the `BLAS=1`?

**Reply by Satyam7166-tech** [reply]
Yes, it is.

<img width="851" alt="image" src="https://github.com/PromtEngineer/localGPT/assets/62897696/f11e5193-a655-4c2e-bfe4-70b2449258ef">

**Comment by Satyam7166-tech** [comment]
@PromtEngineer , Do you have any updates on what I can do to make this utilize mps?

Sorry to bug you. I understand if you have some other matters to attend.

**Comment by zouy50** [comment]
Hi, I'm a beginner either. Your problem really so strange that I am confused too. But I can give you some suggestions for trying:

- you can run this with terminal not in IDE
- you can use the other model try again
- you can run model without this project, such as only run model with [llamacpp](https://github.com/abetlen/llama-cpp-python)
- you can change the documents, such as a simple one
- check your system settings maybe some config make this problem, for example, the power manage, the other application settings affecting

Sorry can't for helping you.

**Reply by Satyam7166-tech** [reply]
Thank you so much for your efforts.

My friend was facing the same error on his mac so I think the problem is with our setup.
Only using llama.cpp was working for us (mps usage) but connecting it to vector database is where the issues began.

Though for the past couple of days, I've been playing around with [Ollama](https://github.com/jmorganca/ollama) and now thankfully, it's working for my use case.
Again, thanks for your efforts.

Do you think I should close the issue now? I am not sure of the protocol.

**Comment by PromtEngineer** [comment]
@Satyam7166-tech The embedding model is also using GPU and since an NVIDIA GPU is not present on your machine, that might slow down the usage of Mac GPU. Let's keep this issue open in case if someone finds a solution.
